### DISCUSSION TITLE
updating a focusedAtom causing re render in another component that has a different focusedAtom, where both of them are derived from an object atom

### DISCUSSION THREAD

**Discussion by shamimfahad** [pmndrs/jotai#1259] [Q&A] [answered]
Hi, let me show you my files before asking the question

atoms.js
```javascript
import { atom } from "jotai";
import { focusAtom } from "jotai/optics";

export const objectAtom = atom({ a: 5, b: 10 });
export const aAtom = focusAtom(objectAtom, (optic) => optic.prop("a"));
export const bAtom = focusAtom(objectAtom, (optic) => optic.prop("b"));
```
App.js
```javascript
import "./styles.css";

import ControlA from "./ControlA";
import ControlB from "./ControlB";

export default function App() {
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <ControlA />
      <ControlB />
    </div>
  );
}
```
ControlA.js
```javascript
iimport { useAtom } from "jotai";

import { aAtom } from "./atoms";

const ControlA = () => {
  const [a, setA] = useAtom(aAtom);
  console.log("a got rendered");
  return (
    <div>
      <span>Value of a: {a}</span>
      <button
        onClick={() => {
          console.log("increment a");
          setA((oldA) => oldA + 1);
        }}
      >
        Increment a
      </button>
    </div>
  );
};

export default ControlA;
```
ControlB.js
```javascript
import { useAtom } from "jotai";

import { bAtom } from "./atoms";

const ControlB = () => {
  const [b, setB] = useAtom(bAtom);
  console.log("b got rendered");

  return (
    <div>
      <span>Value of b: {b}</span>
      <button
        onClick={() => {
          console.log("increment b");
          setB((oldB) => oldB + 1);
        }}
      >
        Increment b
      </button>
    </div>
  );
};

export default ControlB;
```

Now, whenever I increment one of the focusAtoms, both of them gets re rendered, which shouldn't be the case according to the docs. 
![image](https://user-images.githubusercontent.com/48179840/176681372-8f4ee398-c8c9-425f-98bf-7a3106b522a0.png)

Is there anything that I'm doing wrong? or maybe missed something that I should've done? Let me know how I can stop one component from re rendering when updating the other atom. 

you can find the sandbox [here](https://codesandbox.io/s/stupefied-hofstadter-7t5wme?file=/src/App.js)

**Comment by dai-shi** [comment, marked as answer]
It should be the duplicate of #1256.

> It's working as intended. It's not an issue.
> You can wrap `console.log` with `useEffect` to get more intuitive output.
> 
> This is the change in React not in jotai. See this example: https://twitter.com/dai_shi/status/1534170089981100032
> 
> Closing it but this is something others may ask, so it might be a good idea to note something in docs somewhere.
